weaver/core/drivers/fabric-driver/server/satp.ts

```diff
@@ -0,0 +1,363 @@
+    const targetNetwork = performLockRequest2['target-network'];
+    const channel = performLockRequest2['channel'];
+    const chaincodeId = performLockRequest2['chaincode-id'];
+
+    // Hash
+    let hash: HashFunctions.Hash
+    if (hashFn == 'SHA512') {
+        hash = new HashFunctions.SHA512()
+    } else {
+        hash = new HashFunctions.SHA256()
+    }
+
+    if (hashBase64) {
+        hash.setSerializedHashBase64(hashBase64)
+    }
+    else {
+        logger.info(`No hash provided, using random preimage`)
+    }
+
+    // Timeout
+    var timeout = 0, timeout2 = 0;
+    const currTime = Math.floor(Date.now() / 1000);
+    if (performLockRequest2['timeout-epoch']) {
+        let duration = performLockRequest2['timeout-epoch'] - currTime
+        timeout = performLockRequest2['timeout-epoch']
+        timeout2 = performLockRequest2['timeout-epoch'] + duration
+    }
+    else if (performLockRequest2['timeout-duration']) {
+        timeout = currTime + performLockRequest2['timeout-duration']
+        timeout2 = currTime + 2 * performLockRequest2['timeout-duration']
+    }
+
+    let gateway: Gateway = await getNetworkGateway(networkName);
+    const network: Network = await gateway.getNetwork(channel);
+    const contract = network.getContract(chaincodeId);
+
+    const params = performLockRequest2['param'].split(':')
+    const lockerCert = Buffer.from(lockerWalletPath).toString('base64')
+    const recipientCert = Buffer.from(recipientWalletPath).toString('base64')
+
+    var funcToCall, asset
+
+    if (performLockRequest2['fungible']) {
+        funcToCall = SatpAssetManager.createFungibleHTLC
+        asset = 'Fungible Asset'
+    } else {
+        funcToCall = SatpAssetManager.createHTLC
+        asset = 'Asset'
+    }
+
+    console.info(`Asset Lock: Lock ${asset}:\n`);
+    try {
+        console.info(`Trying ${asset} Lock: ${params[0]}, ${params[1]} by ${locker} for ${recipient}`)
+        const res = await funcToCall(contract,
+            params[0],
+            params[1],
+            recipientCert,
+            hash,
+            timeout,
+            null)
+        if (!res.result) {
+            throw new Error()
+        }
+        console.info(`${asset} Locked with Contract Id: ${res.result}, preimage: ${res.hash.getPreimage()}, hashvalue: ${res.hash.getSerializedHashBase64()}`)
+        console.info('Asset has been locked successfully')
+
+    } catch (error) {
+        console.error(`Could not Lock ${asset} in ${targetNetwork}`)
+    }
+
+    await gateway.disconnect()
+    logger.info('Gateways disconnected.')
+
+    const client = getRelayClientForAssetStatusResponse();
+    const request = new satp_pb.SendAssetStatusRequest();
+    request.setSessionId(performLockRequest.getSessionId());
+    request.setStatus("Locked");
+    client.sendAssetStatus(request, relayCallback);
+}
+
+async function createAssetHelper(
+    createAssetRequest: driverPb.CreateAssetRequest,
+    networkName: string
+): Promise<any> {
```

Comment:
Fix return type

Reply:
@outsidethecode Please replace with `unknown` as per the earlier suggestion.

Reply:
Fixed.